data/rao-result/rao-result-json/src/main/java/com/powsybl/openrao/data/raoresultjson/serializers/AngleCnecResultArraySerializer.java

```diff
@@ -87,10 +88,10 @@ private static void serializeAngleCnecResultForOptimizationStateAndUnit(Instant
 
         jsonGenerator.writeObjectFieldStart(serializeUnit(unit));
         if (!Double.isNaN(angle)) {
-            jsonGenerator.writeNumberField(ANGLE, Math.round(100.0 * angle) / 100.0);
+            jsonGenerator.writeNumberField(ANGLE, roundValueBasedOnMargin(angle, margin, 2));
         }
         if (!Double.isNaN(margin)) {
-            jsonGenerator.writeNumberField(MARGIN, Math.round(100.0 * margin) / 100.0);
+            jsonGenerator.writeNumberField(MARGIN, roundValueBasedOnMargin(margin, margin, 2));
```

Comment:
you only need to round flow/angle/voltage values, I'd rather you stayed with the previous formula

Reply:
besides, even though the feature was requested for SWE CNE alone, I believe it is relevant to extend it to RaoResult since the same confusing situation could occur